packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/Services/ReferenceMap.cs

```diff
@@ -58,8 +61,14 @@ ConstructorInfo GetByRefConstructor()
 
                 BindingFlags constructorFlags = BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic;
 
-                return type.GetConstructor(constructorFlags, null, new[] { typeof(ByRefValue) }, null);
+                // Get proxy class implementation for abstract types.
+                if (type.IsClass && type.IsAbstract)
```

Comment:
We already know that the type is a `class`, because `GetClassType` did not return null.

Reply:
I agree that we already know it is a class, since if it was an interface it was already converted to a proxy. Will remove redundant check.